.github/workflows/update-helm-charts-index.yml

```diff
@@ -0,0 +1,53 @@
+name: hashicorp/vault-helm/update-helm-charts-index
+on:
+  push:
+    tags:
+      - 'v[0-9]+\.[0-9]+\.[0-9]+\.*'
+
+jobs:
+  update-helm-charts-index:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: verify Chart version matches tag version
+        run: |-
+          export TAG=${RELEASE_TAG:-${{ github.ref_name }}}
+          git_tag=$(echo "${TAG#v}")
+          chart_tag=$(yq r Chart.yaml version)
+          if [ "${git_tag}" != "${chart_tag}" ]; then
+            echo "chart version (${chart_tag}) did not match git version (${git_tag})"
+            exit 1
+          fi
+        env:
+          RELEASE_TAG: "${{ inputs.release-tag }}"
+      - name: update helm-charts index
+        id: update
+        run: |-
+          export GITHUB_TOKEN="${{secrets.helm_charts_github_token }}"
```

Comment:
I don't think this secret exists as a repository or org secret. Is it an org secret that needs to be shared with this repo?

Reply:
I think you (as the owners) need to create a PAT with your team's service account and add it to the repository's secrets. It needs to have `actions:write` to the `hashicorp/helm-charts` repository. To my knowledge, we don't use org-wide ones (since that'd be shared secrets and all that jazz).

Reply:
Everything else looks good, but we do need a github PAT to be defined in the repo's secrets before this can be merged

Reply:
@swenson added the neccessary secret as `HELM_CHARTS_GITHUB_TOKEN`. I think that resolves this now